cmd/clusterctl/hack/local-overrides.py

```diff
@@ -0,0 +1,184 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+###################
+
+# local-overrides.py takes in input a list of provider and, for each of them, generates the components YAML from the
+# local repositories (the GitHub repositories clone), and finally stores it in the clusterctl local override folder
+
+# prerequisites:
+
+# - the script should be executed from sigs.k8s.io/cluster-api/ by calling cmd/clusterctl/hack/local-overrides.py
+# - there should be a sigs.k8s.io/cluster-api/clusterctl-settings.json file with the list of provider for which
+#   the local overrides should be generated and the list of provider repositories to be included (on top of cluster-api).
+# {
+#    "providers": [ "cluster-api", "kubeadm", "aws"],
+#    "provider_repos": ["../cluster-api-provider-aws"]
+# }
+# - for each additional provider repository there should be a sigs.k8s.io/<provider_repo>/clusterctl-settings.json file e.g.
+# {
+#   "name": "aws",
+#   "config": {
+#     "componentsFile": "infrastructure-components.yaml",
+#     "nextVersion": "v0.5.0",
+#     "type": "InfrastructureProvider"
+# }
+
+###################
+
+import json
+import subprocess
+import os
+import errno
+
+settings = {}
+
+providers = {
+      'cluster-api': {
+              'componentsFile': 'core-components.yaml',
+              'nextVersion': 'v0.3.0',
+              'type': 'CoreProvider',
+      },
+      'kubeadm': {
+            'componentsFile': 'bootstrap-components.yaml',
+            'nextVersion': 'v0.3.0',
+            'type': 'BootstrapProvider',
+            'configFolder': '/bootstrap/kubeadm/config/default',
+      },
+}
+
+def load_settings():
+    global settings
+    try:
+        settings = json.load(open('clusterctl-settings.json'))
+    except  Exception as e:
+        raise Exception('failed to load clusterctl-settings.json: {}'.format(e))
+
+def load_provider():
+    provider_repos = settings.get('provider_repos', [])
+    for repo in provider_repos:
+        file = repo + '/clusterctl-settings.json'
+        try:
+            provider_details = json.load(open(file))
+            provider_name = provider_details['name']
+            provider_config = provider_details['config']
+            provider_config['repo'] = repo
+            providers[provider_name] = provider_config
+        except  Exception as e:
+            raise Exception('failed to load clusterctl-settings.json from repo {}: {}'.format(repo, e))
+
+def kustomize(args):
```

Comment:
This is really a generic command runner. Do you want to hard-code `kustomize` in here and remove it from the invocations below, or rename it to something more generic?

Reply:
done